source/Halibut/Transport/SecureListeningClient.cs

```diff
@@ -67,7 +71,7 @@ public void ExecuteTransaction(ExchangeAction protocolHandler, CancellationToken
                     IConnection connection = null;
                     try
                     {
-                        connection = connectionManager.AcquireConnection(exchangeProtocolBuilder, new TcpConnectionFactory(clientCertificate, asyncHalibutFeature, halibutTimeoutsAndLimits), ServiceEndpoint, log, cancellationToken);
+                        connection = connectionManager.AcquireConnection(exchangeProtocolBuilder, new TcpConnectionFactory(clientCertificate, asyncHalibutFeature, halibutTimeoutsAndLimits, streamFactory), ServiceEndpoint, log, cancellationToken);
```

Comment:
Note that the streamFactory is passed to the ctr of the TcpConnectionFactory

Reply:
We added `streamFactory` to this class because we don't have DI, and it is needed for `TcpConnectionFactory`.

I wonder if instead, we should add the `TcpConnectionFactory` as a constructor dependency instead. This is why I think it makes sense:

- You could get rid of all the arguments needed for `TcpConnectionFactory` as arguments for this class
- `TcpConnectionFactory` doesn't appear to hold any state. So I don't think it needs to be recreated each time.
- It would clean this class up

Reply:
I wasn't doing that to minimise changes in the PR, but that seems to make sense.

Reply:
It looks very nice in the end